applications/scripts/deep_micrograph_cleaner/deep_micrograph_cleaner.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python2
+"""/***************************************************************************
+ *
+ * Authors:    Ruben Sanchez Garcia rsanchez@cnb.csic.es
+ *
+ * CSIC
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.csic.es'
+ ***************************************************************************/
+"""
+
+import os
+import sys
+import xmipp_base
+from xmipp3 import Plugin
+import pyworkflow.em.metadata as md
+
+BAD_IMPORT_MSG='''
+Error, tensorflow/keras is probably not installed. Install it with:\n  ./scipion installb deepLearnigToolkit
+If gpu version of tensorflow desired, install cuda 8.0 or cuda 9.0
+We will try to automatically install cudnn, if unsucesfully, install cudnn and add to LD_LIBRARY_PATH
+add to SCIPION_DIR/config/scipion.conf
+CUDA = True
+CUDA_VERSION = 8.0  or 9.0
+CUDA_HOME = /path/to/cuda-%(CUDA_VERSION)
+CUDA_BIN = %(CUDA_HOME)s/bin
+CUDA_LIB = %(CUDA_HOME)s/lib64
+CUDNN_VERSION = 6 or 7
+'''
+
+class ScriptMicrographCleanerEm(xmipp_base.XmippScript):
+    def __init__(self):
+
+        xmipp_base.XmippScript.__init__(self)
+
+    def getDoubleParamWithDefault(self, paramName, conditionFun= lambda x: False, defaultVal=None):
```

Comment:
This would be good to have in all protocols

Reply:
For protocols we have something similar that @dmaluenda  implemented. Perhaps we can reuse that solution for scripts an pylibs